### PR TITLE
Added Eclipse Che typescript E2E functional tests PR check

### LINKED ISSUE #22044: Restore Eclipse Che typescript E2E functional tests PR check
### Is your task related to a problem? Please describe

There are Eclipse Che typescript E2E functional tests [1] which doesn't have PR check to avoid breaking changes in the main branch.

[1] https://github.com/eclipse/che/tree/main/tests/e2e

### Describe the solution you'd like

Add github action which will be triggered on new commit in PR, and run smoke Eclipse Che typescript E2E functional test.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by nallikaea**
### What does this PR do?
Resolve issue #22044 

- Added GitHub Action that runs Empty Workspace API test, build Che e2e test container and runs Empty Workspace UI test;
- Empty Workspace tests were choosen because of low resource of VM (7RAM, 10Gb disk), other cases returned _'0/1 nodes are available: 1 Insufficient cpu. preemption: 0/1 nodes are available: 1 No preemption victims found for incoming pod..'_
- Updated Dashboard pageobject according to #22184
- Updated quarkus, smoke and empty ws tests to improve stability and avoid hard-coded values
- Added possibility to work with factory urls with query parameters (like '_[https://github.com/quarkus-api-example?df=smoke-test.devfile.yaml](https://github.com/che-incubator/quarkus-api-example?df=smoke-test.devfile.yaml)_') into method getProjectNameFromGitUrl()
- Renamed other actions with understandable names

### Screenshot/screencast of this PR

image
image

### What issues does this PR fix or reference?

Resolve issue #22044 

### How to test this PR?

https://github.com/eclipse/che/actions/runs/5629951250

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [x] [Tests are covering the bugfix or new feature ](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix-or-new-feature)
- [x] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [x] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [x] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

**Review by musienko-maxim — APPROVED** [review]

**Review by ScrewTSW — APPROVED** [review]
LGTM.

Do you have any example run of SmokeTest suite?

File: .github/workflows/next-build.yml [hunk 1]
```diff
@@ -7,7 +7,7 @@
 # SPDX-License-Identifier: EPL-2.0
 #
 
-name: build-next
+name: Build and push Next Che E2E image to quai.io
```
Comment (dmytro-ndp) [inline .github/workflows/next-build.yml]: `build-next` is unified name across Eclipse Che component repos, e.g. https://github.com/eclipse-che/che-server/blob/main/.github/workflows/next-build.yml

So, I would keep it untouched.

File: .github/workflows/pr-check.yml [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+        with:
+          minikube-version: v1.23.2
+
+      - name: Install chectl
+        run: bash <(curl -sL https://www.eclipse.org/che/chectl/) --channel=next
+
+      - name: Deploy Che
+        run: |
+          #
+          # load Che-Code image into minikube
```
Comment (dmytro-ndp) [inline .github/workflows/pr-check.yml:67]: It has to be next version for Eclipse Che Next (`--channel=next` in line 57 above).

Therefore, correct image is address: quay.io/che-incubator/che-code:next .

File: .github/workflows/next-build.yml (hunk 1 above)
Comment (nallikaea) [reply .github/workflows/next-build.yml]: it is just a name which will be shown in Actions tab and has no effect on build actions:

https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#name

I believe it should be clear what has built, why and where was pushed from this name. You don`t need to go through all workflow code to understand that.

image

File: .github/workflows/pr-check.yml [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+          npm run driver-less-test
+
+      - name: Build E2E test docker image
+        run: |
+          set -xe
+
+          cd tests/e2e
+          docker build -t quay.io/eclipse/che-e2e:"${{ env.pr_number }}" -f build/dockerfiles/Dockerfile .
+
+      - name: Run Empty Workspace UI test from che-e2e container
```
Comment (dmytro-ndp) [inline .github/workflows/pr-check.yml:119]: IMHO, it make sense to run SmokeTest instead - it covers project import additionally https://github.com/eclipse/che/blob/main/tests/e2e/specs/SmokeTest.spec.ts

Also, there is a plan to run SmokeTest in PR check of Eclipse Che repos:
- https://github.com/eclipse/che/issues/22046
- https://github.com/eclipse/che/issues/22045
- https://github.com/eclipse/che/issues/22047
- https://github.com/eclipse/che/issues/22101

File: .github/workflows/pr-check.yml (hunk 2 above)
Comment (nallikaea) [reply .github/workflows/pr-check.yml:67]: we tested that test commands still works correctly, not our product. I think we can use stable version here.

File: .github/workflows/pr-check.yml (hunk 3 above)
Comment (nallikaea) [reply .github/workflows/pr-check.yml:119]: Please, read my description

Comment (nallikaea) [reply .github/workflows/pr-check.yml:119]: + we check that changes not broke e2e test functionality, commands from packege.json. We don`t check "che" here.

Comment (nallikaea) [reply .github/workflows/pr-check.yml:119]: https://github.com/eclipse/che/pull/22377#issuecomment-1649304686

File: .github/workflows/pr-check.yml (hunk 2 above)
Comment (dmytro-ndp) [reply .github/workflows/pr-check.yml:67]: It's about checking PRs into the `main` branch, isn't it?
And `main` branch is dedicated to testing Eclipse Che Next.

So, it's reasonable to run PR check against Eclipse Che Next.

File: .github/workflows/pr-check.yml (hunk 3 above)
Comment (dmytro-ndp) [reply .github/workflows/pr-check.yml:119]: Let's comment it somewhere in the repo what exactly does PR check do?

**Review by dmytro-ndp — APPROVED** [review]
Looks commonly good with a few non-critical notices.

File: .github/workflows/pr-check.yml (hunk 2 above)
Comment (nallikaea) [reply .github/workflows/pr-check.yml:67]: okay

File: .github/workflows/next-build.yml (hunk 1 above)
Comment (dmytro-ndp) [reply .github/workflows/next-build.yml]: I see.

My point was that the this eclipse/che repo is not dedicated to e2e tests (looking at the root folder files), so I was not sure if it's eligible to modify `build-next` action.

Moreover - this action has outdated code which we don't need anymore:
```
    - name: Cache local Maven repository
      uses: actions/cache@v3
      with:
       path: ~/.m2/repository
       key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
       restore-keys: |
        ${{ runner.os }}-maven-      
    - name: Set up JDK 11
      uses: actions/setup-java@v3
      with:
        distribution: 'adopt'
        java-version: '11'
```

Anyway, we can consider movement to dedicated repo https://github.com/eclipse-che/che-tests

BTW: there is a TYPO in name: `quai.io` > `quay.io`.

Comment (nallikaea) [reply .github/workflows/next-build.yml]: Fixed

File: .github/workflows/pr-check.yml (hunk 3 above)
Comment (nallikaea) [reply .github/workflows/pr-check.yml:119]: Added

**Pull request merged**